### PR TITLE
Port ws-date-picker from terminal

### PR DESCRIPTION
**Pull request by AlexisDeschamps**
## Why
Use ws-date-picker in other projects like w4w-ui.

## What Changed
Added ws-date-picker component. It was converted from the ws-date-picker directive in terminal.

## Bug/Ticket Tracker
https://trello.com/c/LP91ReAt

## Notes
When webpacked, ws-date-picker does not have access to moment and _ (underscore), which would be in the terminal global scope. Thus, moment and _ were added as globals through webpack, as importing them causes undefined errors in terminal

## To do
Add more component unit tests.

## Screenshot or Gif
screen shot 2017-10-05 at 5 40 39 pm

## Checklist
- [ ] Approved by Dev Manager
- [x] Includes unit/integration tests
- [ ] Completed pre-production checks on https://staging.wealthsimple.com
- [ ] *(After deploy)* Completed post-production checks

### CODE REVIEW

File: src/form/date-picker/ws-date-picker.service.test.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+describe('wsDatePickerService', () => {
+  let wsDatePickerService;
+  let baseTime;
+
+  beforeEach(angular.mock.module(commonModule));
+  beforeEach(angular.mock.inject(['wsDatePickerService', (_service_) => {
+    wsDatePickerService = _service_;
+    baseTime = new Date(2013, 9, 23);
+    jasmine.clock().mockDate(baseTime);
```
Comment (SethDavenport) [inline src/form/date-picker/ws-date-picker.service.test.js]: :+100: nice use of `jasmine.clock`

**Comment by AlexisDeschamps** [comment]
Awesome, OK. I will try removing it.

**Comment by SethDavenport** [comment]
Also, CC @DanGodfrey @ericjim I think you guys were talking in slack about the datepicker recently - can you give @AlexisDeschamps some pointers on what needs to be tested in Terminal with this change?

**Comment by AlexisDeschamps** [comment]
Had to fix the unit tests by adding babel-polyfill to karma.conf.js. Otherwise, array.includes would throw errors (only in the tests). Is there a more convenient way of doing this?

**Comment by SethDavenport** [comment]
IIRC babel-polyfill is just a wrapper around core-js, which is already in our deps.  Can you instead try adding `import 'core-js/es7/array';` to `index.js` instead?

File: src/form/date-picker/ws-date-picker.component.test.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import commonModule from '../..';
+
+describe('wsDatePickerCompoenent', () => {
+  let $compile;
+  let outerScope;
+  let element;
+  // let elementWrapper;
```
Comment (SethDavenport) [inline src/form/date-picker/ws-date-picker.component.test.js]: Comments?

Comment (AlexisDeschamps) [reply src/form/date-picker/ws-date-picker.component.test.js]: Woops, sorry, I am about to add stuff there.

**Comment by AlexisDeschamps** [comment]
Nice, 'core-js/es7/array' worked instantly.

File: package.json [hunk 3]
```diff
@@ -50,6 +51,7 @@
     "babel-plugin-transform-class-properties": "^6.24.1",
     "babel-plugin-transform-object-assign": "^6.8.0",
     "babel-plugin-transform-object-rest-spread": "^6.23.0",
+    "babel-polyfill": "^6.26.0",
```
Comment (SethDavenport) [inline package.json]: This should not be needed any more then.

**Comment by grantjk** [comment]
There's changes being made to the original files in Terminal - let's make sure we're capturing all those changes: https://github.com/wealthsimple/terminal/pull/1889

**Pull request merged**